app/Http/Transformers/PostTransformer.php

```diff
@@ -48,12 +47,18 @@ public function transform(Post $post)
                 'total' => $post->reactions_count,
             ],
             'status' => $post->status,
-            'location' => $post->location,
             'created_at' => $post->created_at->toIso8601String(),
             'updated_at' => $post->updated_at->toIso8601String(),
         ];
 
-        if (Gate::allows('viewAll', $post)) {
+        // If the post isn't anonymous or if this is the owner of the post, return PII information.
+        if ((! filter_var($post->actionModel->anonymous, FILTER_VALIDATE_BOOLEAN) && ! is_null($post->actionModel->anonymous)) || $post->northstar_id === token()->id()) {
```

Comment:
Instead of hard-coding the authorization logic here, let's use the standard policy:

```diff
- $post->northstar_id === token()->id()
+ Gate::allows('viewAll', $post)
```

Reply:
Oh you're right! I didn't use [`Gate::allows('viewAll', $post)`](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/Traits/AuthorizesWithToken.php#L16-L19) here initially because I thought that if a machine (Phoenix) requested, it would still return the `northstar_id` but I forgot that this is using the user auth! Thanks for catching!

Reply:
@DFurnes actually I think I need to change it back to `$post->northstar_id === token()->id()` because the `viewAll` gate allows staff to also see this information and we don't want that.

Reply:
It's okay for staff to be able to see this!

Reply:
oh okay! I'll update the tests for that then :)